pkg/types/conversion/installconfig.go

```diff
@@ -79,11 +82,19 @@ func ConvertNetworking(config *types.InstallConfig) {
 	}
 }
 
-// ConvertBaremetal upconverts deprecated fields in the baremetal
+// convertBaremetal upconverts deprecated fields in the baremetal
 // platform. ProvisioningDHCPExternal has been replaced by setting
 // the ProvisioningNetwork field to "Unmanaged"
-func ConvertBaremetal(config *types.InstallConfig) {
+func convertBaremetal(config *types.InstallConfig) {
 	if config.Platform.BareMetal.DeprecatedProvisioningDHCPExternal == true && config.Platform.BareMetal.ProvisioningNetwork == "" {
 		config.Platform.BareMetal.ProvisioningNetwork = baremetal.UnmanagedProvisioningNetwork
 	}
 }
+
+// convertOpenStack upconverts deprecated fields in the OpenStack platform.
+func convertOpenStack(config *types.InstallConfig) {
+	// LbFloatingIP has been renamed to APIFloatingIP
+	if config.Platform.OpenStack.DeprecatedLbFloatingIP != "" && config.Platform.OpenStack.APIFloatingIP == "" {
```

Comment:
Seems like it should be a fatal error to have both set.  Also, probably worth warn-level logging when you do shuffle the value over, so folks using the old value are more likely to hear about the deprecation and update their generators.

Reply:
this will go in a separate PR for all deprecated values. we have a dedicated jira card for this.